scripts/hotrod-integration-test.sh

```diff
@@ -2,32 +2,83 @@
 
 set -euxf -o pipefail
 
+docker_compose_file="./examples/hotrod/docker-compose.yml"
+platforms="linux/amd64,linux/s390x,linux/ppc64le,linux/arm64"
+
+teardown() {
+  echo "Tearing down..."
+  docker compose -f "$docker_compose_file" down
+}
+trap teardown EXIT
+
 make build-examples GOOS=linux GOARCH=amd64
 make build-examples GOOS=linux GOARCH=s390x
 make build-examples GOOS=linux GOARCH=ppc64le
 make build-examples GOOS=linux GOARCH=arm64
 
-REPO=jaegertracing/example-hotrod
-platforms="linux/amd64,linux/s390x,linux/ppc64le,linux/arm64"
 make prepare-docker-buildx
+make create-baseimg
 
-# build image locally (-l) for integration test
+# Loop through each platform (separated by commas)
+for platform in $(echo "$platforms" | tr ',' ' '); do
+  # Extract the architecture from the platform string
+  arch=${platform##*/}  # Remove everything before the last slash
+  make "build-all-in-one" GOOS=linux GOARCH="${arch}"
+done
+
+# Build image locally (-l) for integration test
 bash scripts/build-upload-a-docker-image.sh -l -c example-hotrod -d examples/hotrod -p "${platforms}"
+bash scripts/build-upload-a-docker-image.sh -l -b -c all-in-one -d cmd/all-in-one -p "${platforms}" -t release
 
-# pass --name example-hotrod so that we can do `docker logs example-hotrod` later
-export CID
-CID=$(docker run -d --name example-hotrod -p 8080:8080 "localhost:5000/${REPO}:${GITHUB_SHA}")
+JAEGER_VERSION=$GITHUB_SHA REGISTRY="localhost:5000/" docker compose -f "$docker_compose_file" up -d
 
 i=0
-while [[ "$(curl -s -o /dev/null -w '%{http_code}' localhost:8080)" != "200" && ${i} -lt 30 ]]; do
+while [[ "$(curl -s -o /dev/null -w '%{http_code}' localhost:8080)" != "200" && $i -lt 30 ]]; do
   sleep 1
   i=$((i+1))
 done
+
 body=$(curl localhost:8080)
 if [[ $body != *"Rides On Demand"* ]]; then
   echo "String \"Rides On Demand\" is not present on the index page"
   exit 1
 fi
-docker rm -f "$CID"
 
-bash scripts/build-upload-a-docker-image.sh -c example-hotrod -d examples/hotrod -p "${platforms}"
+response=$(curl -i -X POST "http://localhost:8080/dispatch?customer=123")
+TRACE_ID=$(echo "$response" | grep -Fi "Traceresponse" | awk '{print $2}' | cut -d '-' -f 2)
```

Comment:
```suggestion
TRACE_ID=$(echo "$response" | grep -Fi "Traceresponse:" | awk '{print $2}' | cut -d '-' -f 2)
```
can you grep with `:`? This would indicate that you're reading a header, not just some content of the response.